### PR TITLE
Adding logic to handle the issue when height/width too small

### LINKED ISSUE #1535: Improve style of metric control
### Description

Having two metrics takes all my page. Metric takes 100% of my screen. This is unusable by itself. 

image

The size must be configurable.

This should look like this by default:

image

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by namnguyen20999**
Resolve https://github.com/Avaiga/taipy/issues/1535

Use the `width` and `height` properties to define dimensions within the layout. However, these properties should only set the width and height if they haven't already been specified in the `layout` object. The layout's own definitions should take precedence over the `width` and `height` properties.

```
from taipy.gui import Gui

value = 50

layout = {
    "width": 500,
    "height": 500,
}

page = """
**The layout has priority over the width and height properties.**
<|{value}|metric|layout={layout}|width=300|height=300|> 

**The width and height properties will be used to define the width and height of the Metric's layout.**
<|{value}|metric|width=300|height=300|>
"""
```

image

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 1]
```diff
@@ -166,7 +166,11 @@ const Metric = (props: MetricProps) => {
     const skelStyle = useMemo(() => ({ ...style, minHeight: "7em" }), [style]);
 
     const layout = useMemo(() => {
-        const layout = { ...baseLayout };
+        const layout = {
+            ...baseLayout,
+            height: baseLayout.height !== undefined ? (baseLayout.height >= 10 ? baseLayout.height : 10) : null,
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx]: Why null why not undefined ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Can you change your code example in the PR to show the new behavior?

**Review by FlorianJacta — COMMENTED** [review]
Can we change width and height directly through the "width" and "height" props of the metric and not through *layout*?

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx]: updated

**Comment by namnguyen20999** [comment]
@FlorianJacta  We can do that, but the width and height in the layout control the CSS applied to the Metric within the container, whereas width and height props control the CSS applied to the Metric's container. 

```
from taipy.gui import Gui

value = 50

layout = {
    "margin": {
        "t": 1,
        "b": 1,
        "l": 1,
        "r": 1
    },
    "height": 300,
    "width": 300
}



page = """
<|{value}|metric|layout={layout}|>
<|{value}|metric|height=300px|width=300px|>
"""
```

You could see the differences here:
Screenshot 2024-08-14 at 13 04 44

**Comment by FlorianJacta** [comment]
> @FlorianJacta We can do that, but the width and height in the layout control the CSS applied to the Metric within the container, whereas width and height props control the CSS applied to the Metric's container.
> 
> ```
> from taipy.gui import Gui
> 
> value = 50
> 
> layout = {
>     "margin": {
>         "t": 1,
>         "b": 1,
>         "l": 1,
>         "r": 1
>     },
>     "height": 300,
>     "width": 300
> }
> 
> 
> 
> page = """
> <|{value}|metric|layout={layout}|>
> <|{value}|metric|height=300px|width=300px|>
> """
> ```
> 
> You could see the differences here: Screenshot 2024-08-14 at 13 04 44

Is this something we can improve? There is an issue to make the metric component more usable in terms of style

**Comment by FredLL-Avaiga** [comment]
If you don't specify anything else in the layout, both example render exactly the same.
I propose that the width and height properties define the width and height values in the layout  (if not defined ie layout definition has precedence) and have no impact on the style.
What do you think @namnguyen20999 @FlorianJacta ?

**Comment by FlorianJacta** [comment]
Why not!

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 2]
```diff
@@ -191,12 +184,11 @@ const Metric = (props: MetricProps) => {
         return layout as Partial<Layout>;
     }, [props.title, props.template, props.template_Dark_, props.template_Light_, theme.palette.mode, baseLayout]);
 
-    const plotConfig = {displaylogo: false}
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:194]: why did you remove this ?

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:194]: by the way it should be declared outside the component

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 3]
```diff
@@ -45,8 +45,6 @@ interface MetricProps extends TaipyBaseProps, TaipyHoverProps {
     testId?: string;
     layout?: string;
     defaultLayout?: string;
-    style?: string;
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:48]: Should this be removed from factory.py and viselements.json ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
there was a decision to remove the plotly logo

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:194]: I've added it back. For a moment I forgot this will display the plotly logo

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 3 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:48]: We don't have it in factory.py and viselements.json at the moment.

**Review by FredLL-Avaiga — APPROVED** [review]
Is the issue for width < 10 not a problem anymore ?

**Comment by namnguyen20999** [comment]
> Is the issue for width < 10 not a problem anymore ?

When the width and height are less than 10, the plotly will revert to its default height of 700px and width of 450px. We just need to indicate it in our docs
